drivers/sensor/lsm6dsl/lsm6dsl_trigger.c

```diff
@@ -119,10 +124,13 @@ int lsm6dsl_init_interrupt(const struct device *dev)
 	/* setup data ready gpio interrupt */
 	drv_data->gpio = device_get_binding(config->irq_dev_name);
 	if (drv_data->gpio == NULL) {
-		LOG_ERR("Cannot get pointer to %s.", config->irq_dev_name);
-		return -EINVAL;
+		LOG_INF("Cannot get pointer to %s.", config->irq_dev_name);
+		drv_data->irq_dt_defined = false;
```

Comment:
I would rather see this as `goto end;` with an `end:` label just before the `return 0;`.

It's easier to understand the code if there's one exit point for success returns.  Having two bypasses everything else with the only effect being that interrupts actually aren't supported.  I could see somebody adding non-interrupt-specific  final updates to the configuration just before returning, without recognizing that one path caused a successful exit elsewhere.

Reply:
ack